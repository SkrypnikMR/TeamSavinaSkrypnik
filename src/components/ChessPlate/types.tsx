import React from 'react';

export interface IDiv {
    background: string;
    width: string;
    height: string;
    display: string;
    flexWrap: string;
    flexDirection: string;
    color:string;
    margin: string;
}
export type TCheckers = {
    id: number;
    key: number;
    isChecked: boolean | null;
    color: string | null;
    onClick: (e: React.MouseEvent) => void;
}
