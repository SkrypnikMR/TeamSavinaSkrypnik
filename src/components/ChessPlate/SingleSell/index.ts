import SingleSell from './SingleSell';

export default SingleSell;
