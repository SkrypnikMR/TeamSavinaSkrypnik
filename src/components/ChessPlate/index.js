import Chessplate from "./ChessPlate";

export default Chessplate