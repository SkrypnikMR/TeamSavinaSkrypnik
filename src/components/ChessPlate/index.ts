import { connect } from 'react-redux';
import { ApplicationState } from '../../Store/types';
import { getCheckerDesk } from '../../Store/game/selectors';

import Chessplate from './ChessPlate';

const mapStateToProps = (state: ApplicationState) => ({
    checker: getCheckerDesk(state),
});

export default connect(mapStateToProps)(Chessplate);
