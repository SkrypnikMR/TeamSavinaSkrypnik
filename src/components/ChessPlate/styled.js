import styled from "styled-components";

export const StChessPlate = styled.div`
    background: white;
    width: 400px;
    height: 400px;
`;