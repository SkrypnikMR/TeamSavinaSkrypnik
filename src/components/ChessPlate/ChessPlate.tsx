import React from 'react';
import { TCheckers } from 'src/Store/game/types';
import SingleSell from './SingleSell';
import { StChessPlate } from './styled';

const Chessplate: React.FC<TCheckers> = ({ checker }) => {
    console.log(checker);
    const onHandleClick = (e:React.MouseEvent) => {
        console.log(e);
    };
    return (
        <StChessPlate>
            {items.map(item => <SingleSell id={item} key={item} />,
            )}
        </StChessPlate>
    );
};

export default Chessplate;
