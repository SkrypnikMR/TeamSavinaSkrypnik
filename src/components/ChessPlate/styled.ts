import styled from 'styled-components';

import { IDiv } from './types';

export const StChessPlate = styled.div < IDiv > `
    background: white;
    width: 600px;
    height: 600px;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row-reverse;
    color: gray; //temporary
    margin: 10px auto;
    box-shadow: 0 1px 4px rgba(0, 0, 0, 1),
        -23px 0 20px -23px rgba(0, 0, 0, .8),
        23px 0 20px -23px rgba(0, 0, 0, .8),
        0 0 40px rgba(0, 0, 0, .1) inset;
`;
