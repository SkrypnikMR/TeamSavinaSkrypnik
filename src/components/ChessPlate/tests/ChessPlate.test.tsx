import React from 'react';
import { shallow, mount} from 'enzyme';

import ChessPlate from '../ChessPlate';

describe('ChessPlate', () => {
    it('Should match snapshot', () => {
        const component = shallow(<ChessPlate />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StAddRoom', () => {
        const component = mount(<ChessPlate/>);
        expect(component.find('styled__StChessPlate')).toHaveLength(1);
    });
});
