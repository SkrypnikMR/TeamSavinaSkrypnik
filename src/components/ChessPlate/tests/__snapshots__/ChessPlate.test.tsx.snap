// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChessPlate Should match snapshot 1`] = `"<div class=\\"styled__StChessPlate-sc-1ga4euq-0 etIfCT\\"><div id=\\"64\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">64</div><div id=\\"63\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">63</div><div id=\\"62\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">62</div><div id=\\"61\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">61</div><div id=\\"60\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">60</div><div id=\\"59\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">59</div><div id=\\"58\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">58</div><div id=\\"57\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">57</div><div id=\\"56\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">56</div><div id=\\"55\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">55</div><div id=\\"54\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">54</div><div id=\\"53\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">53</div><div id=\\"52\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">52</div><div id=\\"51\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">51</div><div id=\\"50\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">50</div><div id=\\"49\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">49</div><div id=\\"48\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">48</div><div id=\\"47\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">47</div><div id=\\"46\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">46</div><div id=\\"45\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">45</div><div id=\\"44\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">44</div><div id=\\"43\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">43</div><div id=\\"42\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">42</div><div id=\\"41\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">41</div><div id=\\"40\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">40</div><div id=\\"39\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">39</div><div id=\\"38\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">38</div><div id=\\"37\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">37</div><div id=\\"36\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">36</div><div id=\\"35\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">35</div><div id=\\"34\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">34</div><div id=\\"33\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">33</div><div id=\\"32\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">32</div><div id=\\"31\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">31</div><div id=\\"30\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">30</div><div id=\\"29\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">29</div><div id=\\"28\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">28</div><div id=\\"27\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">27</div><div id=\\"26\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">26</div><div id=\\"25\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">25</div><div id=\\"24\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">24</div><div id=\\"23\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">23</div><div id=\\"22\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">22</div><div id=\\"21\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">21</div><div id=\\"20\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">20</div><div id=\\"19\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">19</div><div id=\\"18\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">18</div><div id=\\"17\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">17</div><div id=\\"16\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">16</div><div id=\\"15\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">15</div><div id=\\"14\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">14</div><div id=\\"13\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">13</div><div id=\\"12\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">12</div><div id=\\"11\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">11</div><div id=\\"10\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">10</div><div id=\\"9\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">9</div><div id=\\"8\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">8</div><div id=\\"7\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">7</div><div id=\\"6\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">6</div><div id=\\"5\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">5</div><div id=\\"4\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">4</div><div id=\\"3\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">3</div><div id=\\"2\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">2</div><div id=\\"1\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">1</div></div>"`;
